### ISSUE TITLE
Colons in strings are not parsed correctly in YAML files

### ISSUE THREAD

**Issue by dnicolson** [sbarex/SourceCodeSyntaxHighlight#241]
The following YAML:
```
AWSTemplateFormatVersion: "2010-09-09"
Metadata:
    Generator: "former2"
Description: ""
Resources:
    CloudWatchAlarm:
        Type: "AWS::CloudWatch::Alarm"
        Properties:
            AlarmName: "Alarm"
```

Results in:
<img width="720" alt="Screenshot 2023-05-19 at 21 14 38" src="https://github.com/sbarex/SourceCodeSyntaxHighlight/assets/2276355/905f973a-aa07-42fe-95a4-33b924333006">

**Comment by hylowaker** [comment]
How did you resolve that? I'm also getting the same issue.

<img width="1333" alt="스크린샷 2023-08-03 오후 12 01 52" src="https://github.com/sbarex/SourceCodeSyntaxHighlight/assets/10094075/6e4cabfc-eb78-4d42-8ee8-13c1e2f717fd">

**Comment by dnicolson** [comment]
The issue was fixed in version 2.1.11.

Your issue appears to be different, rather than colons it seems that a combination of letters and version numbers causes the problem.

<img width="596" alt="Screenshot 2023-08-03 at 16 59 45" src="https://github.com/sbarex/SourceCodeSyntaxHighlight/assets/2276355/034a1019-5baa-4df0-a77e-ea185ccb83f2">
